spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/apache/RibbonApacheHttpResponse.java

```diff
@@ -0,0 +1,151 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.util.Collection;
+import java.util.Map;
+
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.reflect.TypeToken;
+import com.netflix.client.ClientException;
+import com.netflix.client.http.CaseInsensitiveMultiMap;
+import com.netflix.client.http.HttpHeaders;
+
+/**
+ * @author Christian Lohmann
+ */
+public class RibbonApacheHttpResponse implements com.netflix.client.http.HttpResponse {
+
+	private HttpResponse httpResponse;
+	private URI uri;
+
+	public RibbonApacheHttpResponse(final HttpResponse httpResponse, final URI uri) {
+		this.httpResponse = httpResponse;
+		this.uri = uri;
+	}
+
+	@Override
+	public Object getPayload() throws ClientException {
+		try {
+			return this.httpResponse.getEntity().getContent();
+		}
+		catch (final IOException e) {
+			throw new ClientException(e.getMessage(), e);
+		}
+	}
+
+	@Override
+	public boolean hasPayload() {
+		return this.httpResponse.getEntity() != null;
+	}
+
+	@Override
+	public boolean isSuccess() {
+		return this.httpResponse.getStatusLine().getStatusCode() == 200;
+	}
+
+	@Override
+	public URI getRequestedURI() {
+		return this.uri;
+	}
+
+	@Override
+	public int getStatus() {
+		return httpResponse.getStatusLine().getStatusCode();
+	}
+
+	@Override
+	public String getStatusLine() {
+		return httpResponse.getStatusLine().toString();
+	}
+
+	@Override
+	public Map<String, Collection<String>> getHeaders() {
+		final Map<String, Collection<String>> headers = Maps.newHashMap();
+		for (final Header header : this.httpResponse.getAllHeaders()) {
+			if (headers.containsKey(header.getName())) {
+				headers.get(header.getName()).add(header.getValue());
+			}
+			else {
+				headers.put(header.getName(), Lists.newArrayList(header.getValue()));
+			}
+		}
+
+		return headers;
+	}
+
+	@Override
+	public HttpHeaders getHttpHeaders() {
+		final CaseInsensitiveMultiMap headers = new CaseInsensitiveMultiMap();
+		for (final Header header : httpResponse.getAllHeaders()) {
+			headers.addHeader(header.getName(), header.getValue());
+		}
+
+		return headers;
+	}
+
+	@Override
+	public void close() {
+		if (this.httpResponse != null && this.httpResponse.getEntity() != null) {
+			try {
+				this.httpResponse.getEntity().getContent().close();
+			}
+			catch (final IOException e) {
+				throw new RuntimeException(e.getMessage(), e);
+			}
+		}
+
+	}
+
+	@Override
+	public InputStream getInputStream() {
+		try {
+			return this.httpResponse.getEntity().getContent();
+		}
+		catch (final IOException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		}
+	}
+
+	@Override
+	public boolean hasEntity() {
+		return hasPayload();
+	}
+
+	@Override
+	public <T> T getEntity(final Class<T> type) throws Exception {
+		return null;
```

Comment:
This is no problem because they are not in use, but a bit ugly. As described above these methods have their origin in ```com.netflix.client.http.HttpResponse``` which i am forced to implement.